swift-sdk/Internal/EmbeddedMessagingManager.swift

```diff
@@ -148,5 +148,5 @@ class EmbeddedMessagingManager: NSObject, IterableEmbeddedMessagingManagerProtoc
     
     private var messages: [IterableEmbeddedMessage] = []
     
-    private var listeners: [IterableEmbeddedMessagingUpdateDelegate] = []
+    private var listeners: NSHashTable<IterableEmbeddedMessagingUpdateDelegate> = NSHashTable(options: [.weakMemory])
 }
```

Comment:
Is there a reason for using weak memory?

Reply:
So if the UI object is removed from memory, we don't inadvertently keep it around (memory leak). We don't have a need to have a direct reference to anything that goes in there.